src/main/java/org/sasanlabs/service/vulnerability/pathTraversal/PathTraversalVulnerability.java

```diff
@@ -41,31 +41,35 @@ public class PathTraversalVulnerability {
             LogManager.getLogger(PathTraversalVulnerability.class);
 
     private static final String URL_PARAM_KEY = "fileName";
-
     private ResponseEntity<GenericVulnerabilityResponseBean<String>> readFile(
             Supplier<Boolean> condition, String fileName) {
-        if (condition.get()) {
-            InputStream infoFileStream =
-                    this.getClass().getResourceAsStream("/scripts/PathTraversal/" + fileName);
-            if (infoFileStream != null) {
-                try (BufferedReader reader =
-                        new BufferedReader(new InputStreamReader(infoFileStream))) {
-                    String information = reader.readLine();
-                    StringBuilder payload = new StringBuilder();
-                    while (information != null) {
-                        payload.append(information);
-                        information = reader.readLine();
-                    }
-                    return new ResponseEntity<GenericVulnerabilityResponseBean<String>>(
-                            new GenericVulnerabilityResponseBean<>(payload.toString(), true),
-                            HttpStatus.OK);
-                } catch (IOException e) {
-                    LOGGER.error("Following error occurred: ", e);
-                }
+        if (!condition.get()) {
+            return ResponseEntity.badRequest().body(new GenericVulnerabilityResponseBean<>("Invalid request condition", false));
+        }
+
+        // Preventing Path Traversal and ensuring only allowed filenames are processed.
+        if (fileName.contains("..") || !ALLOWED_FILE_NAMES.contains(fileName)) {
```

Comment:
Why this change? we wanted to introduce vulnerability in the application and this will disallow ".." in file name.